Lib/test/datetimetester.py

```diff
@@ -3740,6 +3800,83 @@ def test_compat_unpickle(self):
                     derived = loads(data, encoding='latin1')
                     self.assertEqual(derived, expected)
 
+    def test_strptime(self):
+        string = '13:02:47.197'
+        format = '%H:%M:%S.%f'
+        expected = _strptime._strptime_datetime_time(self.theclass, string,
+                                                     format)
+        got = self.theclass.strptime(string, format)
+        self.assertEqual(expected, got)
+        self.assertIs(type(expected), self.theclass)
+        self.assertIs(type(got), self.theclass)
+
+        # bpo-34482: Check that surrogates are handled properly.
+        inputs = [
+            ('13:02\ud80047.197', '%H:%M\ud800%S.%f'),
+            ('13\ud80002:47.197', '%H\ud800%M:%S.%f'),
+        ]
+        for string, format in inputs:
+            with self.subTest(string=string, format=format):
+                expected = _strptime._strptime_datetime_time(self.theclass,
+                                                             string, format)
+                got = self.theclass.strptime(string, format)
+                self.assertEqual(expected, got)
+
+        strptime = self.theclass.strptime
+        self.assertEqual(strptime("+0002", "%z").utcoffset(), 2 * MINUTE)
+        self.assertEqual(strptime("-0002", "%z").utcoffset(), -2 * MINUTE)
+        self.assertEqual(
+            strptime("-00:02:01.000003", "%z").utcoffset(),
+            -timedelta(minutes=2, seconds=1, microseconds=3)
+        )
+        # Only local timezone and UTC are supported
+        for tzseconds, tzname in ((0, 'UTC'), (0, 'GMT'),
+                                 (-_time.timezone, _time.tzname[0])):
+            if tzseconds < 0:
+                sign = '-'
+                seconds = -tzseconds
+            else:
+                sign ='+'
+                seconds = tzseconds
+            hours, minutes = divmod(seconds//60, 60)
+            tstr = "{}{:02d}{:02d} {}".format(sign, hours, minutes, tzname)
+            t = strptime(tstr, "%z %Z")
+            self.assertEqual(t.utcoffset(), timedelta(seconds=tzseconds))
+            self.assertEqual(t.tzname(), tzname)
+
+        # Can produce inconsistent time
+        tstr, fmt = "+1234 UTC", "%z %Z"
+        t = strptime(tstr, fmt)
+        self.assertEqual(t.utcoffset(), 12 * HOUR + 34 * MINUTE)
+        self.assertEqual(t.tzname(), 'UTC')
+        # yet will roundtrip
+        self.assertEqual(t.strftime(fmt), tstr)
+
+        # Produce naive time if no %z is provided
+        self.assertEqual(strptime("UTC", "%Z").tzinfo, None)
+
+        with self.assertRaises(ValueError): strptime("-2400", "%z")
+        with self.assertRaises(ValueError): strptime("-000", "%z")
+        with self.assertRaises(ValueError): strptime("z", "%z")
+
```

Comment:
Similar to above, we should test to see that there are no date components in the string.

```suggestion

def test_strptime_date_with_time_error(self):
    formats = [
        ('2004', '%Y'),
        ('02', '%m'),
        ('13', '%d'),
        ('2004-02-13', '%Y-%m-%d'),
        ('2004-02-03 04:11:23.145', '%Y-%m-%d %H:%M:%S.%f'),
        ('2004-02 12', '%Y-%m %H'),
    ]

    for dtstr, format in formats:
        with self.subTest(format=format):
            with self.assertRaises(ValueError):
                self.theclass.strptime(dtstr, format)
```